rust_src/src/buffers.rs

```diff
@@ -367,4 +373,25 @@ pub fn set_buffer(buffer_or_name: LispObject) -> LispObject {
     buffer
 }
 
+/// Signal a `buffer-read-only' error if the current buffer is read-only.
+/// If the text under POSITION (which defaults to point) has the
+/// `inhibit-read-only' text property set, the error will not be raised.
+#[lisp_fn(min = "0")]
+pub fn barf_if_buffer_read_only(position: Option<EmacsInt>) -> () {
+    let pos = match position {
+        None => point(),
+        Some(n) => n,
+    };
```

Comment:
This could be position.unwrap_or_else(point), but it's fine whichever way you think is clearer.